src/sentry/rules/processor.py

```diff
@@ -89,18 +99,42 @@ def apply_rule(self, rule):
 
         state = self.get_state()
 
+        condition_list = []
+        filter_list = []
+        for rule_cond in rule_condition_list:
+            if self.get_condition_category(rule_cond) == RuleCategory.CONDITION:
+                condition_list.append(rule_cond)
+            else:
+                filter_list.append(rule_cond)
+
+        if not condition_list:
+            return
+
         condition_iter = (self.condition_matches(c, state, rule) for c in condition_list)
+        filter_iter = (self.condition_matches(f, state, rule) for f in filter_list)
+
+        if condition_match == "all":
+            condition_passed = all(condition_iter)
+        elif condition_match == "any":
+            condition_passed = any(condition_iter)
+        elif condition_match == "none":
+            condition_passed = not any(condition_iter)
+        else:
+            self.logger.error("Unsupported action_match %r for rule %d", condition_match, rule.id)
+            return
```

Comment:
We could probably consolidate the logic to convert a match type into a function callable in a function and make this repeated code simpler.

Something like 
```
condition_func = match_to_func(condition_match)
if condition_func:
    condition_passed = condition_func(condition_iter)
else:
   # log error and return
```
Names aren't great here, but something along those lines. 

Also wondering if we should just early exit if conditions or filters don't pass, since we just return a few lines down anyway.

Reply:
Good call

and yea I realize we should definitely return early for the case in which conditions don't pass